moveit_core/online_signal_smoothing/test/test_acceleration_filter.cpp

```diff
@@ -0,0 +1,196 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2024, PickNik Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include <gtest/gtest.h>
+#include <moveit/online_signal_smoothing/acceleration_filter.h>
+#include <moveit/utils/robot_model_test_utils.h>
+
+class AccelerationFilterTest : public testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    robot_model_ = moveit::core::loadTestingRobotModel("panda");
+  };
+
+  void set_limit(std::optional<Eigen::VectorXd> acceleration_limit)
+  {
+    const std::vector<moveit::core::JointModel*> joint_models = robot_model_->getJointModels();
+    auto joint_model_group = robot_model_->getJointModelGroup(move_group_name_);
+    size_t ind = 0;
+    for (auto& joint_model : joint_models)
+    {
+      if (!joint_model_group->hasJointModel(joint_model->getName()))
+      {
+        continue;
+      }
+      std::vector<moveit_msgs::msg::JointLimits> joint_bounds_msg(joint_model->getVariableBoundsMsg());
+      for (auto& joint_bound : joint_bounds_msg)
+      {
+        joint_bound.has_acceleration_limits = acceleration_limit.has_value();
```

Comment:
I think the way you're using this, you can actually make this check just once at the top of the function and then set the same individual `has_acceleration_limits` value for all joints.

Reply:
I don't understand completely what you mean. I can make a variable to copy acceleration_limit.has_value()` but I don't think that will help.